Geometry/VeryForwardGeometryBuilder/interface/CTPPSGeometry.h

```diff
@@ -0,0 +1,137 @@
+/****************************************************************************
+*
+* Authors:
+*  Jan Kašpar (jan.kaspar@gmail.com)
+*
+****************************************************************************/
+
+#ifndef Geometry_VeryForwardGeometryBuilder_CTPPSGeometry
+#define Geometry_VeryForwardGeometryBuilder_CTPPSGeometry
+
+#include "DataFormats/CTPPSDetId/interface/CTPPSDetId.h"
+
+#include "Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h"
+#include "Geometry/VeryForwardGeometryBuilder/interface/CTPPSDDDNames.h"
+
+#include "CLHEP/Vector/ThreeVector.h"
+#include "CLHEP/Vector/Rotation.h"
+
+#include <map>
+#include <set>
+
+/**
+ * \brief The manager class for TOTEM RP geometry.
+ *
+ * This is kind of "public relation class" for the tree structure of DetGeomDesc. It provides convenient interface to
+ * answer frequently asked questions about the geometry of TOTEM Roman Pots. These questions are of type:\n
+ *  - What is the geometry (shift, roatation, material, etc.) of detector with id xxx?\n
+ *  - If RP ID is xxx, which are the sensorss inside this pot?\n
+ *  - If hit position in local detector coordinate system is xxx, what is the hit position in global c.s.?\n
+ * etc. (see the comments in definition bellow)
+ **/
+
+class CTPPSGeometry
+{
+  public:
+    typedef std::map<unsigned int, const DetGeomDesc* > mapType;
+    typedef std::map<int, const DetGeomDesc* > RPDeviceMapType;
+    typedef std::map<unsigned int, std::set<unsigned int> > mapSetType;
+
+    CTPPSGeometry() {}
+    ~CTPPSGeometry() {}
+
+    /// build up from DetGeomDesc
+    CTPPSGeometry(const DetGeomDesc * gd)
+    {
+      build(gd);
```

Comment:
Hi @ianna 
Indeed... changed `build` return type to `void` (will consider throwing exceptions in the future if needed)